kamon-statsd/src/main/scala/kamon/statsd/StatsDMetricsSender.scala

```diff
@@ -48,7 +48,7 @@ class StatsDMetricsSender(remote: InetSocketAddress, maxPacketSizeInBytes: Long,
   }
 
   def writeMetricsToRemote(tick: TickMetricSnapshot, udpSender: ActorRef): Unit = {
-    val packetBuilder = new MetricDataPacketBuilder(maxPacketSizeInBytes, udpSender, remote)
+    val packetBuilder = new MetricDataPacketBuilder(maxPacketSizeInBytes, udpSender, statsDHost, statsDPort)
```

Comment:
You could simply make remote a def in this actor using the provided host/port internally and keep the MetricDataPacketBuilder intact.